Add LSK token

sameersubudhi opened the pull request:
<!-- Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md -->

**Description**

This PR adds the Lisk (LSK) token to the Superchain Token List on Ethereum and Lisk networks.

**Token Addresses**:

- Ethereum: [0x6033F7f88332B8db6ad452B7C6D5bB643990aE3f](https://eth.blockscout.com/address/0x6033F7f88332B8db6ad452B7C6D5bB643990aE3f)
- Lisk: [0xac485391EB2d7D88253a7F1eF18C37f4242D1A24](https://blockscout.lisk.com/address/0xac485391EB2d7D88253a7F1eF18C37f4242D1A24)
- Sepolia: [0x16B840bA01e2b05fc2268eAf6d18892a11EC29D6](https://eth-sepolia.blockscout.com/address/0x16B840bA01e2b05fc2268eAf6d18892a11EC29D6)
- Lisk Sepolia Testnet: [0x8a21CF9Ba08Ae709D64Cb25AfAA951183EC9FF6D](https://sepolia-blockscout.lisk.com/address/0x8a21CF9Ba08Ae709D64Cb25AfAA951183EC9FF6D)

**Official sources for contract addresses:**
- **Mainnet**: https://github.com/LiskHQ/lisk-contracts/blob/main/deployment/addresses/mainnet/addresses.md
- **Sepolia**: https://github.com/LiskHQ/lisk-contracts/blob/main/deployment/addresses/sepolia/addresses.md

**Tests**

Not Applicable; entry validated along with the changes in https://github.com/ethereum-optimism/ethereum-optimism.github.io/pull/897.

**Additional context**

- This PR is dependent on the changes in https://github.com/ethereum-optimism/ethereum-optimism.github.io/pull/897.
- I will rebase this PR once the above PR is merged.

hamdiallam commented:
hey @sameersubudhi, this PR just needs to be rebased now that your prior PR was merged.

Also seeing `nobridge` being set to true but the [contract](https://blockscout.lisk.com/address/0xac485391EB2d7D88253a7F1eF18C37f4242D1A24?tab=contract) still being an OptimismMintableERC20.

It looks like the contract isn't upgradable. Why not use the OptimismMintableERC20Factory to create the L2 representation and removing the `nobridge` attribute?

sameersubudhi commented:
Hello @hamdiallam,

Thank you for pointing out the mistake with `nobridge` being set to `true`. I've now rebased the PR and fixed the `nobridge` issue.

Regarding the L2 token contracts (mainnet: [0xac485391EB2d7D88253a7F1eF18C37f4242D1A24](https://blockscout.lisk.com/address/0xac485391EB2d7D88253a7F1eF18C37f4242D1A24), sepolia: [0x8a21CF9Ba08Ae709D64Cb25AfAA951183EC9FF6D](https://sepolia-blockscout.lisk.com/address/0x8a21CF9Ba08Ae709D64Cb25AfAA951183EC9FF6D)) we did not directly use the `OptimismMintableERC20Factory` contract to deploy these because it is currently lacking the support for `ERC20Permit`. So, we implemented our own token contract to add support for the following interfaces: `IOptimismMintableERC20, ERC20, ERC20Permit`.

I hope this answers your query.
Please feel free to drop me a question in case you need any further clarifications.
